src/Scrutor/ServiceCollectionExtensions.Composite.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static partial class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Registers <typeparamref name="TConcrete"/> as a composite that wraps all
+        /// existing registrations of <typeparamref name="TInterface"/>.
+        /// </summary>
+        /// <typeparam name="TInterface"></typeparam>
+        /// <typeparam name="TConcrete"></typeparam>
+        /// <param name="services"></param>
+        public static void AddComposite<TInterface, TConcrete>(this IServiceCollection services)
+          where TInterface : class
+          where TConcrete : class, TInterface
+        {
+            var wrappedDescriptors = services.Where(s => s.ServiceType == typeof(TInterface)).ToList();
+            foreach (var descriptor in wrappedDescriptors)
+                services.Remove(descriptor);
+
+            var objectFactory = ActivatorUtilities.CreateFactory(
+              typeof(TConcrete),
+              new[] { typeof(IEnumerable<TInterface>) });
+
+            var maxWrappedServiceLifetime = wrappedDescriptors
+                .Select(d => d.Lifetime)
+                .DefaultIfEmpty(ServiceLifetime.Scoped)
+                .Max();
+
+            services.Add(ServiceDescriptor.Describe(
+              typeof(TInterface),
+              s => (TInterface)objectFactory(s, new[] { wrappedDescriptors.Select(d => s.CreateInstance(d)).Cast<TInterface>() }),
+              maxWrappedServiceLifetime)
+            );
+        }
+
+        private static object CreateInstance(this IServiceProvider services, ServiceDescriptor descriptor)
```

Comment:
This looks an awful lot like the existing `GetInstance` method. Can we reuse it?

Reply:
Yeah, it looks the same to me.  Currently `GetInstance` is a private in `ServiceCollectionExtensions.Decoration.cs`; I suggest moving it to a new `ServiceCollectionExtensions.cs` for shared helpers.

Sound good?

Reply:
Or leave it where it is I guess - they're all in the same shared class!